PyRoute/Galaxy.py

```diff
@@ -300,6 +302,11 @@ def read_sectors(self, sectors, pop_code, ru_calc):
                     continue
                 star = Star.parse_line_into_star(line, sec, pop_code, ru_calc)
                 if star:
+                    star.index = star_counter
+                    star_counter += 1
+                    if star not in sec.worlds:
```

Comment:
Have you encountered a case where a world is being added twice in the same sector? 

I'm not sure why you have this check.

Reply:
I have encountered this in pathological cases, usually where some wombat inadvertently doubled a star up.  Bit before, don't want to get bit again, etc.

Reply:
To explain further, @tjoneslo :
An invariant of the integer-index work in this PR is that the mapping is **bijective** .
Each index in the mapping must be mapped to **exactly one**, **distinct** Star object.
Each Star object must be mapped to a likewise unique index.

In less formal terms, each direction of the mapping **must** cleanly round-trip back to where it started.

I think integer-indexes has exposed this problem - previously, using Star objects made node addition to galaxy.stars idempotent.

Take a given Star, S that has somehow gotten duplicated.
First time S is added, its index is (say) 120, so that S.index is set to 120, and star_mapping[120] = S
Second time the Star gets added, its index is (say) 130, so S.index becomes 130, and star_mapping[130] = S.

star_mapping[120] does _not_ round trip, so we have an invariant violation.

Reply:
The problem is the system may be treating S1 (at index 120) and S2 (at index 130) as distinct stars, if overlapping in position. It's almost certainly the case someone has done a copy/paste incorrectly. But we should be scrubbing the data here, not attempting to guess which of the two system has been updated correctly.

Reply:
Well, as such duplication breaks the bijectiveness of the mapping, I've promoted that check to an assertion.  If a star has already been added to the sector before, print a message saying star X is duplicated in sector Y, then cough and die.